### PR TITLE
expression: implement vectorized evaluation for builtinFromDaysSig

### PR DESCRIPTION
**Pull request by mmyj**
### What problem does this PR solve? 
implement vectorized evaluation for builtinFromDaysSig, for #12101
### What is changed and how it works?
```
 go test -v -benchmem -bench=BenchmarkVectorizedBuiltinTimeFunc -run=BenchmarkVectorizedBuiltinTimeFunc -args 'builtinFromDaysSig'
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinTimeFuncGenerated-4           1000000000               0.00682 ns/op         0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinFromDaysSig-VecBuiltinFunc-4             55994             21838 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinFromDaysSig-NonVecBuiltinFunc-4          28222             41867 ns/op               0 B/op          0 allocs/op
PASS
ok      github.com/pingcap/tidb/expression      3.133s
```
### Check List 
Tests 
- Unit test

### CODE REVIEW

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -481,11 +481,31 @@ func (b *builtinTimeStringTimeDiffSig) vecEvalDuration(input *chunk.Chunk, resul
...
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ResizeTime(n, false)
+	result.MergeNulls(buf)
+	t64s := result.Times()
```
Comment (XuHuaiyu) [inline expression/builtin_time_vec.go]: s/ t64s/ `ts` or `times` or `rs`

File: expression/builtin_time_vec.go [hunk 2]
```diff
@@ -481,11 +481,31 @@ func (b *builtinTimeStringTimeDiffSig) vecEvalDuration(input *chunk.Chunk, resul
...
 
 func (b *builtinFromDaysSig) vecEvalTime(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf); err != nil {
```
Comment (XuHuaiyu) [inline expression/builtin_time_vec.go]: s/ `:=` / `=`

File: expression/builtin_time_vec.go (hunk 1 above)
Comment (mmyj) [reply expression/builtin_time_vec.go]: I will fix it.

File: expression/builtin_time_vec.go (hunk 2 above)
Comment (mmyj) [reply expression/builtin_time_vec.go]: I will fix it.

**Comment by ngaut** [comment]
PTAL @zimulala

**Pull request merged**